Provider/src/main/java/dev/openfeature/contrib/providers/ConfidenceFeatureProvider.kt

```diff
@@ -54,14 +58,43 @@ class ConfidenceFeatureProvider private constructor(
             eventsPublisher.publish(OpenFeatureEvents.ProviderReady)
         }
     }
+
     override fun initialize(initialContext: EvaluationContext?) {
         if (initialContext == null) return
+
+        // refresh cache with the last stored data
+        storage.read()?.let(cache::refresh)
```

Comment:
Shall we add some error handling here in case storage data is corrupted?

Reply:
Devils advocate: What can we do? And how can we detect a corrupt storage data?

Reply:
If we're not able to read the storage for some reason when it's not null, we should handle that - perhaps by wiping storage?

Reply:
not being able to read the storage is typically an OS error and it's rare but technically can happen. I believe wiping the data wouldn't solve the problem that OS failed to read the data. wdyt?

Reply:
Makes sense to me! Will mark as resolved 👍